plutus-tx/src/PlutusTx/Bool.hs

```diff
@@ -17,7 +17,7 @@ import Prelude (Bool (..), otherwise)
 --
 infixr 3 &&
 (&&) :: Bool -> Bool -> Bool
-(&&) l r = if l then r else False
+(&&) l ~r = if l then r else False
```

Comment:
Yes, since && tends to get inlined. This matters a lot for at least one test case.

Reply:
Actually, I'm back to not understanding how changing the strictness of a function parameter can ever make a difference, even if it's inlined. Function application is strict, and inlining should not change that, otherwise the inliner is broken! Can we write a specific test case that shows what it does?

Reply:
See added Note